### PR TITLE
editoast: utoipa for /studies

### PR DESCRIPTION
**Pull request by Tristramg**
refs #4580

### CODE REVIEW

File: editoast/src/models/study.rs [hunk 1]
```diff
@@ -42,42 +44,56 @@ use super::List;
...
     pub last_modification: NaiveDateTime,
     #[diesel(deserialize_as = Option<NaiveDate>)]
+    #[schema(value_type = Option<NaiveDate>)]
     pub start_date: Option<Option<NaiveDate>>,
     #[diesel(deserialize_as = Option<NaiveDate>)]
+    #[schema(value_type = Option<NaiveDateTime>)]
     pub expected_end_date: Option<Option<NaiveDate>>,
     #[diesel(deserialize_as = Option<NaiveDate>)]
+    #[schema(value_type = Option<NaiveDateTime>)]
     pub actual_end_date: Option<Option<NaiveDate>>,
```
Comment (leovalais) [inline editoast/src/models/study.rs]: `NaiveDate{,Time}` mixing here

File: editoast/src/views/mod.rs [hunk 2]
```diff
@@ -76,6 +76,7 @@ schemas! {
     documents::schemas(),
     pathfinding::schemas(),
     projects::schemas(),
+    study::schemas(),
```
Comment (leovalais) [inline editoast/src/views/mod.rs]: You can include the schemas of `studies` in the schema! declaration of projects like so:

```rust
crate::schemas! {
    PaginatedResponseOfProjectWithStudies,
    ProjectCreateForm,
    ProjectPatchForm,
    study::schemas(), // <====
}
```

File: front/src/applications/operationalStudies/views/Project.tsx [hunk 3]
```diff
@@ -118,7 +113,7 @@ export default function Project() {
         };
         try {
           const filteredData = await postSearch(payload).unwrap();
-          let filteredStudies = [...filteredData] as SearchStudyResult[];
```
Comment (leovalais) [inline front/src/applications/operationalStudies/views/Project.tsx]: Note: this conflicts with #5464

File: editoast/openapi_legacy.yaml
Comment (leovalais) [inline editoast/openapi_legacy.yaml:1]: You forgot to remove the unused schemas as well. I use this script to list them:

```bash
cat openapi_legacy.yaml | yq '.components.schemas | keys' | awk '{print $2}' | while read -r schema; do
    grep "#/components/schemas/$schema\"" openapi_legacy.yaml >/dev/null || echo $schema
done
```

```
StudyUpsertRequest
StudyResult
```

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks! Looks fine to me, just a few minor changes. I still have to test it.

**Comment by leovalais** [comment]
Tested!

File: editoast/openapi_legacy.yaml
Comment (Tristramg) [reply editoast/openapi_legacy.yaml:1]: Nice query! And thank you, that was missing to bring the diff to a negative number ;)

File: front/src/applications/operationalStudies/views/Project.tsx (hunk 3 above)
Comment (Tristramg) [reply front/src/applications/operationalStudies/views/Project.tsx]: I’ll wait for the merge to handle the conflict

**Comment by Tristramg** [comment]
Thank you!

**Review by clarani — APPROVED** [review]
Tested, LGTM :white_check_mark:

**Review by leovalais — APPROVED** [review]
Thanks! 😄

**Pull request merged**